students/brian_warn/session04/kata_test.py

```diff
@@ -0,0 +1,94 @@
+#!/Library/Frameworks/Python.framework/Versions/3.6/bin/python
+# Kata Fourteen
+import random
+
+
+def main():
+    exceptions = ['.', '?', ' ', '!', ',', '-']
+
+    def build_trigram(infile):
+        ''' Function that builds the trigram dictionary from the source file. '''
+
+        trigrams = {}
+        with open(infile) as sherlock:
+            sherlock.readline()
+            for line in sherlock:
+                line = line.rstrip('\r\n')
+                word = line.split(" ")
+                for i in range(len(word) - 2):
+                    pair = tuple(word[i:i + 2])
+                    follower = word[i + 2]
+                    if follower not in exceptions:
+                        pass
+                    # Find dictionary method that does this in one call
+                    if pair in trigrams:
+                        trigrams[pair].append(follower)
+                    if pair not in trigrams:
+                        trigrams[tuple(pair)] = [follower]
+
+        # Check to see what's in trigrams dictionary:
+        # items = trigrams.items()
+        # for x in items:
+        #     print(x)
+        return trigrams.items()
```

Comment:
I'm pretty sure you want to return the entire trigram dict....

Reply:
This shows an area where I hadn't worked through the next step of the sentence-building task.  My thought at this point was that I just needed the key in order to go through `build_trigram` again to get the next word.

Reply:
What .items() does is return both the keys and the values as an iterable (kind of like a list). So you have all the information, but have lost the ability to efficiently look up a particular word pair.

And looking up a word pair is exactly what you need to to to build a new text.